### PR TITLE
 [DOCS] Reformats cluster stats API and expands common params

### PR DESCRIPTION
**Pull request by szabosteve**
Relates to elastic/docs#937 and https://github.com/elastic/elasticsearch/issues/45197.

This PR updates the cluster stats API to align with the new [API reference template](https://github.com/elastic/docs/blob/master/shared/api-ref-ex.asciidoc) and adds a new param to the common params list (flat_settings).

Resources:
* [Cluster stats API spec](https://github.com/elastic/elasticsearch/blob/master/rest-api-spec/src/main/resources/rest-api-spec/api/cluster.stats.json)
* [Cluster stats doc](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/cluster-stats.html)

### CODE REVIEW

**Comment by szabosteve** [comment]
@jrodewig If you want me to add a response body section, I'll do it in a separate PR as https://github.com/elastic/elasticsearch/pull/45206 depends on the changes that are contained in this PR. Adding the response body section will be time-consuming, I wouldn't like to delay the merge of another PR until it is done.

File: docs/reference/cluster/stats.asciidoc [hunk 1]
```diff
@@ -1,10 +1,44 @@
...
+The Cluster Stats API allows to retrieve statistics from a cluster wide 
+perspective. The API returns basic index metrics (shard numbers, store size, 
+memory usage) and information about the current nodes that form the cluster 
+(number, roles, os, jvm versions, memory usage, cpu and installed plugins).
+
+
+[[cluster-stats-api-path-params]]
+==== {api-path-parms-title}
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=local]
```
Comment (jrodewig) [inline docs/reference/cluster/stats.asciidoc]:
```suggestion
```

**Review by jrodewig — APPROVED** [review]
LGTM. Made one suggestion to remove a copy/paste error.

I don't think you need to include a response body ATM. I can handle that as a separate project if needed.

Thanks @szabosteve.

**Comment by szabosteve** [comment]
@jrodewig Thanks! If you need help in the response body project, please ping me I'd be happy to be involved.

**Review by jrodewig — APPROVED** [review]
LGTM.

**Comment by szabosteve** [comment]
Backported to 7.x and 7.3.

**Pull request merged**